providers/edge/src/airflow/providers/edge/cli/edge_command.py

```diff
@@ -365,14 +367,34 @@ def check_running_jobs(self) -> None:
 
     def heartbeat(self) -> None:
         """Report liveness state of worker to central site with stats."""
-        state = (
-            (EdgeWorkerState.TERMINATING if _EdgeWorkerCli.drain else EdgeWorkerState.RUNNING)
-            if self.jobs
-            else EdgeWorkerState.IDLE
-        )
+        if _EdgeWorkerCli.drain:
+            state = EdgeWorkerState.TERMINATING
+        elif self.jobs:
+            if _EdgeWorkerCli.maintenance_mode:
+                state = EdgeWorkerState.MAINTENANCE_PENDING
+            else:
+                state = EdgeWorkerState.RUNNING
+        else:
+            if _EdgeWorkerCli.maintenance_mode:
+                state = EdgeWorkerState.MAINTENANCE_MODE
+            else:
+                state = EdgeWorkerState.IDLE
         sysinfo = self._get_sysinfo()
         try:
-            self.queues = worker_set_state(self.hostname, state, len(self.jobs), self.queues, sysinfo)
+            worker_info = worker_set_state(self.hostname, state, len(self.jobs), self.queues, sysinfo)
+            self.queues = worker_info["queues"] if isinstance(worker_info["queues"], list) else None
+            if worker_info["state"] in (
+                EdgeWorkerState.MAINTENANCE_REQUEST,
+                EdgeWorkerState.MAINTENANCE_PENDING,
+                EdgeWorkerState.MAINTENANCE_MODE,
+            ):
+                if not _EdgeWorkerCli.maintenance_mode:
+                    logger.info("Maintenance mode requested!")
+                _EdgeWorkerCli.maintenance_mode = True
```

Comment:
As the maintenance mode is passed via the heartbeat, it takes some time until is picked up __and__ also some time to report back the current state. Does it make sense to change the timing of heartbeat such that a confirmation/change in maintenance is signalled immediately and not in the next cycle?